src/py/flwr/server/state/state.py

```diff
@@ -134,5 +134,9 @@ def unregister_node(self, node_id: int) -> None:
         """Remove `node_id` from state."""
 
     @abc.abstractmethod
-    def get_nodes(self) -> Set[int]:
+    def get_nodes(self, workload_id: str) -> Set[int]:
         """Retrieve all currently stored node IDs as a set."""
+
+    @abc.abstractmethod
+    def create_workload(self) -> str:
+        """Create one workload."""
```

Comment:
Similar to what I mentioned above, we should write a spec here that clearly states who is responsible for `workload_id` generation and what type the `workload_id` should have.

Reply:
I would prefer IDs to be generated by the state and not outside of the state. I think for legacy reasons, in the case of the `node_id`, we generate it outside of the state.

Reply:
As discussed offline, we will move ID generation into the `State` in a subsequent PR (and so so for all IDs, not just the `workload_id`)